getting_started/scripting/gdscript/static_typing.rst

```diff
@@ -0,0 +1,413 @@
+2. Core classes and nodes (``Object``, ``Node``, ``Area2D``,
+   ``Camera2D``, etc.)
+3. Your own, custom classes. Look at the new
+   `class_name <https://godot.readthedocs.io/en/latest/getting_started/step_by_step/scripting_continued.html#register-scripts-as-classes>`__
+   feature to register types in the editor.
+
+.. note::
+
+    You don't need to write type hints for constants as Godot sets it automatically from the assigned value. But you can still do so to make the intent of your code clearer.
+
+Custom Variable Types
+~~~~~~~~~~~~~~~~~~~~~
+
+You can use any class, including your custom classes, as types. There
+are two ways to use them in scripts. The first method is to preload the
+script you want to use as a type in a constant:
+
+::
+
+    const Rifle = preload('res://player/weapons/Rifle.gd')
+    var my_rifle : Rifle
+
+The second method is to use the ``class_name`` keyword when you create.
+For the example above, your Rifle.gd would look like this:
+
+::
+
+    extends Node2D
+    class_name Rifle
+
+If you use ``class_name``, Godot registers the Rifle type globally in
+the editor, and you can use it anywhere without having to preload it
+into a constant:
+
+::
+
+    var my_rifle : Rifle
+
+Variable Casting
+~~~~~~~~~~~~~~~~
+
+Type casting is a key concept in typed languages. We call the conversion
+of a value from from one type of another casting.
+
+Imagine an Enemy in your game, that ``extends Area2D``. You want it to
+collide with the Player, a ``KinematicBody2D`` with a script called
+``PlayerController`` attached to it. You use the ``on_body_entered``
+signal to detect the collision. With typed code, the body you detect is
+going to be a generic ``PhysicsBody2D``, and not your
+``PlayerController`` on the ``_on_body_entered`` callback.
+
+You can check if this ``PhysicsBody2D`` is your Player with the ``as``
+casting keyword, and using the colon ``:`` again to force the variable
+to use this type. This forces the variable to stick to the
+``PlayerController`` type:
+
+::
+
+    func _on_body_entered(body : PhysicsBody2D) -> void:
+    var player := body as PlayerController
+    if not player:
+        return
+    player.damage()
+
+As we’re dealing with a custom type, if the ``body`` doesn’t extend
+``PlayerController``, the ``player``\ variable will be set to ``null``.
+We can use this to check if the body is the player or not. We will also
+get full autocompletion on the player variable thanks to that cast.
+
+.. note::
+
+    If you try to cast with a built-in type and it fails, Godot will throw an error.
+
+Safe lines
+^^^^^^^^^^
+
+You can also use casting is to ensure safe lines. Safe lines are a new
+tool in Godot 3.1 to tell you when ambiguous lines of code are
+type-safe. As you can mix and match typed and dynamic code, at times,
+Godot doesn’t have enough information to if an instruction will trigger
+an error or not at runtime.
```

Comment:
I think you forgot to remove "is" from "You can also use casting is to ensure safe lines."